lang/py/avro/schema.py

```diff
@@ -469,10 +511,19 @@ def __str__(self):
         return json.dumps(self.to_json())
 
     def to_json(self, names=None):
-        if names is None:
-            names = Names()
+        names = names or Names()
+
         to_dump = self.props.copy()
         to_dump['type'] = self.type.to_json(names)
+
+        return to_dump
+
+    def to_canonical_json(self, names=None):
+        names = names or Names()
```

Comment:
At some point (not here) we should consider tightening up these interfaces so we don't have to copy patterns like this into every implementation.

Reply:
Amen. 👍 
There are more things I have noted, but I want to get the main package to be on par with the specification first before I go after them.